test/Table_Tests/src/Database/Postgres_Spec.enso

```diff
@@ -338,6 +341,96 @@ postgres_specific_spec connection db_name setup =
+            local_dt = "2022-05-04 15:30:00"
+            table.at "A" . cast Value_Type.Char . to_vector . should_equal [local_dt, local_dt, local_dt, local_dt]
+
+            # Then when downloaded, it should be interpreted at the 'system default' timezone.
+            materialized_table = table.read
+            materialized_table.at "A" . to_vector . should_equal [dt1, dt1, dt1, dt1]
+
+            # The Inexact_Type_Coercion warning is silenced for this case:
+            Problems.assume_no_problems materialized_table
+
+            # We also check how the timestamp column behaves with interpolations:
+            v.each my_dt-> Test.with_clue my_dt.to_text+": " <|
+                t2 = table.filter "A" (Filter_Condition.Equal to=my_dt)
+                is_in_system_tz = my_dt.at_zone Time_Zone.system . time_of_day == my_dt.time_of_day
+                ## Unfortunately, this will work in the following way:
+                   - if the date-time represented as local time in the current system default timezone is 15:30,
+                     then it will match _all_ entries (as they do not have a timezone).
+                   - otherwise, the date-time is converted into UTC before being passed to the Database,
+                     and only then the timezone is stripped - so the local time is actually shifted.
+
+                   This is not ideal - ideally we'd want the local date time to be extracted from the timestamp directly,
+                   before any date conversions happen - this way in _all_ 4 cases we would get 15:30 local time
+                   and all rows would always be matching.
+
+                   That logic is applied when uploading a table. However, in custom queries, we do not currently have
+                   enough metadata to infer that the Date_Time that is being passed to a given `?` hole in the query
+                   should be treated as a local date-time or a zoned date-time.
+                   Thus, we pass it as zoned by default to avoid losing information - and that triggers the conversion
+                   on the Database side. If we want to change that, we would need to add metadata within our operations,
+                   so that an operation like `==` will infer the expected type of the `?` hole based on the type of the
+                   second operand.
+                case is_in_system_tz of
+                    True ->
+                        my_dt.at_zone Time_Zone.system . time_of_day . to_display_text . should_equal "15:30:00"
+                        t2.row_count . should_equal 4
+                        t2.at "A" . to_vector . should_equal [dt1, dt1, dt1, dt1]
+                    False ->
+                        my_dt.at_zone Time_Zone.system . time_of_day . to_display_text . should_not_equal "15:30:00"
+                        t2.row_count . should_equal 0
+                        t2.at "A" . to_vector . should_equal []
```

Comment:
This is far from ideal, because in _interpolations_ (i.e. operations like `Filter_Condition.Equal`) the `Date_Time` is by default treated as `timestamptz`. But that means it is converted using different logic than when uploading data into the Database.

1. When uploading, we have special logic that ensures that the `Date_Time` has timezone stripped, but the local-date-time it represents is preserved as-is. This is something **we want**, because it seems to be the only intuitive invariant we can keep for converting Date_Time with timezone to one without timezone.
2. When interpolating, we currently cannot have this special logic - because our operations don't know enough information about the type the `?` 'hole' should have. So it just uses whatever _value_ is in there - but the Enso `Date_Time` value _does have a timezone_. So it is sent as `timestamptz` and the standard Postgres conversion happens - which first converts it to UTC and _then_ gets the local-date-time from it - possibly shifting the local-date-time and leading to surprising behaviour.

Reply:
Possible solutions:
1. Revert the upload logic. We could rely on the default DB logic of converting to UTC and _then_ stripping the TZ. **I don't think we should do that**, as we discussed with @jdunkerley this is rather un-intuitive.
2. Make our operations more 'smart' about the expected types. We could 'infer' that if one side of `==` has a database value of `timestamp without timezone` then the other side that is a `?` interpolation, should also be treated as such - detecting these scenarios and adding the special conversion when preparing the query. This is rather complex to implement and will likely only benefit this specific edge case. Still, we may not be able to properly infer the type in some cases.
3. We can improve our toolset allowing to more easily work with these values, without changing the default semantics.
  i. We probably should add a warning when comparing a `timestamp without timezone` with `timestamp with timezone` column or Enso value.
  ii. We should have an operation that would allow to add a timezone to a `timestamp without timezone` column, i.e. `no_tz_col.date_ops.at_zone Time_Zone.utc` which would return a `timestamptz` column with the local-date-time interpreted to be UTC.


IMHO (3) would be our best choice.

Reply:
I added #8122 as a ticket to implement (3.i.) because I think we really need a warning in this case - otherwise users working with `timestamp without timezone` columns will be getting weird behaviour without explanation.

We can add another ticket for (3.ii.) if we decide we want it.

Reply:
Also added #8126 for the (3.ii.) case. We can discuss on next backlog review if we actually want to implement it, but it seems like the most likely solution.